crates/ruma-common/src/push/condition.rs

```diff
@@ -451,7 +526,7 @@ mod tests {
         assert!("foo bar".matches_pattern("foo", true));
         assert!("Foo bar".matches_pattern("foo", true));
         assert!(!"foobar".matches_pattern("foo", true));
-        assert!(!"foo bar".matches_pattern("foo*", true));
```

Comment:
Why was this here with no comment of technical non-compliance? No blocking note, i just wanna note that in the future, we should at least `FIXME` cases like this.